server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1616,6 +1615,50 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }, listener::onFailure);
     }
 
+    /**
+     * Updates the repository generation that running deletes and snapshot finalizations will be based on for this repository if any such
+     * operations are found in the cluster state while setting the safe repository generation.
+     *
+     * @param state  cluster state to update
+     * @param oldGen previous safe repository generation
+     * @param newGen new safe repository generation
+     * @return updated cluster state
+     */
+    private ClusterState updateRepositoryGenerations(ClusterState state, long oldGen, long newGen) {
```

Comment:
It's somewhat questionable to even continue tracking the generation of an operation at this point since it's always equal to the safe repository generation (except for the corner case of having just mounted the repository maybe ...). I added the generation updating here for now to keep the change-set small but I'd look into removing the generation from deletes and snapshots in a follow-up (technically easy ... but BwC of that change is tricky).